### PR TITLE
feat(table): convert column mgmt demos to fullscreen

### LINKED ISSUE #7357: Feat(Table): Full screen demo for Column Management Modal

### PR DESCRIPTION
**Pull request by jenny-s51**
**What**: Closes #7357

### CODE REVIEW

**Review by mcarrano — CHANGES_REQUESTED** [review]
@jenny-s51 This looks great, but there seems to be a bug with column reordering via drag & drop. If I reorder columns and then click Save, upon returning to the table, it's empty.

**Review by jenny-s51 — COMMENTED** [review]
Great catch @mcarrano ! I've been thoroughly investigating here; this is not a trivial bug to fix, so I opened #7471 to address this issue as part of a necessary refactor. 

Our demo is using a deprecated API on `DataList` that implemented the drag/drop functionality. It would make more sense to refactor the code here in favor of the new `DragDrop` component, which will likely fix this bug and also keep us up to date with how we currently demo `DragDrop` in [DataList](https://github.com/patternfly/patternfly-react/blob/18817c800714e4d743cca5c8d4eca01b5da00616/packages/react-core/src/components/DataList/examples/DataListDraggable.tsx#L64)! WDYT @tlabaj ?

**Comment by mmenestr** [comment]
Looks good, apart from what @mcarrano pointed out! Another thing, that maybe is outside the scope of this issue, is that I expected that while I'm dragging a row around, the other rows would move to "make room" for the dragged row, so that I would be able to have a visual of what i'm doing before I drop the row where I want it. Idk if that makes sense, but I just feel like it's hard to tell where the row is going to land while i'm dragging it..

**Comment by jenny-s51** [comment]
Good point @mmenestr - the visual look of the current drag/drop implementation here is different from the drag/drop visuals that we see in the DataList draggable demo. This is also due to the newer `DragDrop` component that's being used there! 

The dragging visuals will be addressed separately in #7471 when this code is refactored to use `DragDrop`, which will align the appearance with the draggable Data list demo!

**Comment by tlabaj** [comment]
> Great catch @mcarrano ! I've been thoroughly investigating here; this is not a trivial bug to fix, so I opened #7471 to address this issue as part of a necessary refactor.
> 
> Our demo is using a deprecated API on `DataList` that implemented the drag/drop functionality. It would make more sense to refactor the code here in favor of the new `DragDrop` component, which will likely fix this bug and also keep us up to date with how we currently demo `DragDrop` in [DataList](https://github.com/patternfly/patternfly-react/blob/18817c800714e4d743cca5c8d4eca01b5da00616/packages/react-core/src/components/DataList/examples/DataListDraggable.tsx#L64)! WDYT @tlabaj ?

@mcarrano @jenny-s51, I agree, we would want to suggest refactoring to use new drag and drop.

**Comment by mmenestr** [comment]
kk sounds good! Just tried it out again, and the drag and drop is working as expected now!! But one thing I noticed is that when I go back into the manage columns modal after I've re-ordered something, the order that the columns show up in does not match the order they're in inside the table. Shouldn't the order of the columns in the modal, match the order of the columns in the table?

**Comment by jenny-s51** [comment]
@mmenestr that will also be addressed as part of the refactor in #7471! The way the rows in the modal are currently ordered/implemented can be fixed/updated once we use `DragDrop` here instead 👍

File: packages/react-table/src/demos/Table.md [hunk 1]
```diff
@@ -1661,24 +1673,30 @@ class ColumnManagementAction extends React.Component {
...
+          <PageSection padding={{ default: 'noPadding', xl: 'padding' }}>
+            <Card component="div">
+              <Table
+                gridBreakPoint="grid-xl"
+                header={
+                  <React.Fragment>
+                    <Toolbar id="page-layout-table-column-management-action-toolbar-top">{toolbarItems}</Toolbar>
+                  </React.Fragment>
+                }
+                aria-label="This is a table with checkboxes"
```
Comment (thatblindgeye) [inline packages/react-table/src/demos/Table.md:1686]:
```suggestion
                aria-label="This is a table with draggable column management"
```

File: packages/react-table/src/demos/Table.md [hunk 2]
```diff
@@ -898,24 +901,30 @@ class ColumnManagementAction extends React.Component {
...
```
Comment (thatblindgeye) [inline packages/react-table/src/demos/Table.md:914]:
```suggestion
                aria-label="This is a table with column management"
```

**Review by thatblindgeye — CHANGES_REQUESTED** [review]
Had a couple of nits below, but I'm also noticing an issue with both of the examples updated. The Pagination looks to be breaking outside of the main page content when I scroll to the very bottom of the page:

Pagination breaking out of page

versus how the bulk select demo looks:

Pagination in bulk select demo

**Pull request closed without merging**